### PR TITLE
Adds event batcher for the indexer

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/24713|

## Description

Adds the implementation for the batcher in charge of sending events from the comms API to the Indexer.

### CODE REVIEW

File: framework/wazuh/core/batcher/client.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        Returns
+        -------
+        int
+            Unique identifier assigned to the event.
+        """
+        assigned_uid = id(event)
+        self.queue.send_to_mux(assigned_uid, event)
+        return assigned_uid
+
+
```
Comment (fdalmaup) [inline framework/wazuh/core/batcher/client.py:40]:
```suggestion

```

File: framework/wazuh/core/batcher/tests/test_timer.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import pytest
+import asyncio
```
[the remark this answers is not in the document]
Comment (RamosFe) [reply framework/wazuh/core/batcher/tests/test_timer.py]: Fixed in 2df43eb2a6541c8cd9b41db1381deb47fc58b3ba.

File: framework/wazuh/core/batcher/client.py (hunk 1 above)
Comment (RamosFe) [reply framework/wazuh/core/batcher/client.py]: Fixed in 2df43eb2a6541c8cd9b41db1381deb47fc58b3ba.

File: framework/wazuh/core/batcher/tests/test_client.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+@patch("builtins.id")
+def test_send_event(id_mock, queue_mock):
+    """Check that the `send_event` method works as expected."""
+    batcher = BatcherClient(queue=queue_mock)
+
+    event = {"data": "test event"}
+    expected_uid = 1234
+
+    id_mock.return_value = expected_uid
+    queue_mock.send_to_mux.return_value = expected_uid
```
[the remark this answers is not in the document]
Comment (RamosFe) [reply framework/wazuh/core/batcher/tests/test_client.py]: Fixed in 2df43eb2a6541c8cd9b41db1381deb47fc58b3ba.

File: framework/wazuh/core/batcher/batcher.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    """Class to execute the batching in a separate process.
+
+    Parameters
+    ----------
+    queue : MuxDemuxQueue
+        MuxDemuxQueue instance for message multiplexing and demultiplexing.
+    config : BatcherConfig
+        Configuration parameters for batching, such as maximum elements and size.
+    """
+    def __init__(self, queue: MuxDemuxQueue, config: BatcherConfig):
```
[the remark this answers is not in the document]
Comment (RamosFe) [reply framework/wazuh/core/batcher/batcher.py]: Fixed in 2df43eb2a6541c8cd9b41db1381deb47fc58b3ba.

File: framework/wazuh/core/batcher/batcher.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    """Manage the batching of messages from a MuxDemuxQueue and send them in bulk to an indexer.
+
+    Parameters
+    ----------
+    queue : MuxDemuxQueue
+        MuxDemuxQueue instance for message multiplexing and demultiplexing.
+    config : BatcherConfig
+        Configuration parameters for batching, such as maximum elements and size.
+    """
+    def __init__(self, queue: MuxDemuxQueue, config: BatcherConfig):
```
[the remark this answers is not in the document]
Comment (RamosFe) [reply framework/wazuh/core/batcher/batcher.py]: Fixed in 2df43eb2a6541c8cd9b41db1381deb47fc58b3ba.

**Review by fdalmaup — APPROVED** [review]
LGTM!

**Pull request merged**